types/rlp_unmarshal_storage.go

```diff
@@ -105,13 +95,13 @@ func (t *Transaction) unmarshalStoreRLPFrom(p *fastrlp.Parser, v *fastrlp.Value)
 			return err
 		}
 
-		t.SetTransactionType(tType)
-
 		elems = elems[1:]
+
+		t.InitInnerData(tType)
```

Comment:
Maybe it is a bit more logical to swap these two calls:

```suggestion
		t.InitInnerData(tType)

        elems = elems[1:]
```